File: backend/app/routes/audit.py
"""Audit API routes."""

from fastapi import APIRouter, HTTPException
from app.audit.explainer import generate_audit_trail

router = APIRouter(tags=["Audit"])


@router.get("/audit/{invoice_no}")
async def get_audit_trail(invoice_no: str):
    """Generate explainable audit trail for an invoice."""
    try:
        trail = generate_audit_trail(invoice_no)
        if "error" in trail:
            raise HTTPException(status_code=404, detail=trail["error"])
        return trail
    except HTTPException:
        raise
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))

File: backend/src/routes/audit.ts
import { Router } from 'express';
import { generateAuditTrail } from '../audit/explainer';

const router = Router();

router.get('/:invoiceNo', async (req, res) => {
    try {
        const result = await generateAuditTrail(req.params.invoiceNo);
        res.json(result);
    } catch (error: any) {
        if (error.message.includes('not found')) {
            res.status(404).json({ error: error.message });
        } else {
            res.status(500).json({ error: error.message });
        }
    }
});

export default router;
